doc/user-guide/windowing.md

```diff
@@ -0,0 +1,187 @@
+
+```text
+1, 5, 10, 15, 20, 25, 30, 35, 40
+|--|                                [0  - 4]
+   |--|                             [5  - 9]
+      |---|                         [10 - 14]
+          |---|                     [15 - 19]
+              |---|                 [20 - 24]
+                  |---|             [25 - 29]
+                      |---|         [30 - 34]
+                          |---|     [35 - 39]
+```
+
+#### Sliding Windows
+
+In contrast to fixed windows, sliding windows allow extents to overlap. When a sliding window is specified, we have to give it a range for which the window spans, and a *slide* value for how long to wait between spawning a new window extent. Every data point will fall into exactly `range / slide` number of window extents. We draw out what this looks like for a sliding window with range `15` and slide `5`:
+
+```text
+1, 5, 10, 15, 20, 25, 30, 35, 40
+|---------|                         [0  - 14]
+   |----------|                     [5  - 19]
+      |-----------|                 [10 - 24]
+          |-----------|             [15 - 29]
+              |-----------|         [20 - 34]
+                  |-----------|     [25 - 39]
+```
+
+### Units
+
+Onyx allows you to specify range and slide values in different magnitudes of units, so long as the units can be coverted to the same unit in the end. For example, you can specify the range in minutes, and the slide in seconds. Any value that requires units takes a vector of two elemenets. The first element represents the value, and the second the unit. For example, window specifications denoting range and slide might look like:
+
+```clojure
+{:window/range [1 :minute]
+ :window/slide [30 :seconds]}
+```
+
+See the information model for all supported units. You can use a singular form (e.g. `:minute`) instead of the plural (e.g. `:minutes`) where it makes sense for readability.
+
+Onyx is also capable of sliding by `:elements`. This is often referred to as "slide-by-tuple" in research. Onyx doesn't require a time-based range and slide value. Any totally ordered value will work equivalently.
+
+### Aggregation
+
+Windows allow you accrete data over time. Sometimes, you want to store all the data. Othertimes you want to incrementally compact the data. Window specifications must provide a `:window/aggregation` key. We'll go over an example of every type of aggregation that Onyx supports.
+
```

Comment:
Cool. Thanks for bringing this up, I would have totally missed it otherwise. I got carried away with hammering out the implementation.